### PR TITLE
fix(eslint-config): Remove deprecated jsx-a11y/label-has-for rule

### PR DESCRIPTION
**Pull request by deminoth**
## Description

The rule `jsx-a11y/label-has-for` is deprecated and replaced with `jsx-a11y/label-has-associated-control`.

https://github.com/jsx-eslint/eslint-plugin-jsx-a11y/releases/tag/v6.1.0

### CODE REVIEW

**Comment by herecydev** [comment]
@madalynrose probably interested in this change

**Comment by herecydev** [comment]
Apologies @deminoth I readded this in: #24920 

Looks like it was removed for the recommended version but not the strict, which is annoying 😒. It might be useful to put a comment on the rules to say it's been purposely removed?

File: packages/gatsby/src/utils/eslint-config.ts [hunk 1]
```diff
@@ -89,7 +89,6 @@ export const eslintConfig = (schema: GraphQLSchema): CLIEngine.Options => {
             ],
           },
         ],
-        "jsx-a11y/label-has-for": `warn`,
         "jsx-a11y/label-has-associated-control": `warn`,
```
Comment (madalynrose) [inline packages/gatsby/src/utils/eslint-config.ts:92]:
```suggestion
        //"jsx-a11y/label-has-associated-control": `warn`, was deprecated and replaced with jsx-a11y/has-associated-control in v6.1.0
```

**Comment by herecydev** [comment]
LGTM!

**Pull request merged**